<launch>

    <arg name="robot" default='0' doc="The robot to use."/>
    <include file="$(find nuturtle_robot)/launch/basic_remote.launch">
        <arg name="robot" value="$(arg robot)" />
    </include>

    <rosparam command="load" file="$(find nuturtle_robot)/config/real_waypoints.yaml" />

    <rosparam command="load" file="$(find nuturtle_description)/config/diff_params.yaml" />
    
    <!-- Turtlebot encoder branch -->

    <node machine="turtlebot" pkg="nuturtle_robot" type="turtle_interface" name="turtle_interface" />


    <param name="robot_description" command="xacro '$(find nuturtle_description)/urdf/diff_drive.urdf.xacro'" />

    <node machine="turtlebot" name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
        <remap from="joint_states" to="joint_state"/>
    </node>

    <group if="$(eval arg('robot') != 0)">
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find nuturtle_description)/config/view_odom.rviz"/>
    </group>

    <node machine="turtlebot" name="odometer" pkg="rigid2d" type="odometer" output="screen">
        <param name="~body_frame_id_" type="str" value="base_link" />
        <param name="~odom_frame_id_" type="str" value="odom" />
    </node>


    <!-- waypoints controller -->
    <node machine="turtlebot" name="real_waypoint" pkg="nuturtle_robot" type="real_waypoint" output="screen"/>

</launch>